Sources/Bot/Services/MergeService.swift

```diff
@@ -506,37 +508,64 @@ extension MergeService {
             }
         }
 
-        static func initial(with integrationLabel: PullRequest.Label, statusChecksTimeout: TimeInterval) -> State {
-            return State(integrationLabel: integrationLabel, statusChecksTimeout: statusChecksTimeout, pullRequests: [], status: .starting)
+        static func initial(integrationLabel: PullRequest.Label, topPriorityLabels: [PullRequest.Label], statusChecksTimeout: TimeInterval) -> State {
+            return State(
+                integrationLabel: integrationLabel,
+                topPriorityLabels: topPriorityLabels,
+                statusChecksTimeout: statusChecksTimeout,
+                pullRequests: [],
+                status: .starting
+            )
         }
 
-        init(integrationLabel: PullRequest.Label, statusChecksTimeout: TimeInterval, pullRequests: [PullRequest], status: Status) {
+        init(
+            integrationLabel: PullRequest.Label,
+            topPriorityLabels: [PullRequest.Label],
+            statusChecksTimeout: TimeInterval,
+            pullRequests: [PullRequest],
+            status: Status
+        ) {
             self.integrationLabel = integrationLabel
+            self.topPriorityLabels = topPriorityLabels
             self.statusChecksTimeout = statusChecksTimeout
             self.pullRequests = pullRequests
             self.status = status
         }
 
         func with(status: Status) -> State {
-            return State(integrationLabel: integrationLabel, statusChecksTimeout: statusChecksTimeout, pullRequests: pullRequests, status: status)
+            return State(
+                integrationLabel: integrationLabel,
+                topPriorityLabels: topPriorityLabels,
+                statusChecksTimeout: statusChecksTimeout,
+                pullRequests: pullRequests,
+                status: status
+            )
         }
 
         func include(pullRequests pullRequestsToInclude: [PullRequest]) -> State {
+            let onlyNewPRs = pullRequestsToInclude.filter {
+                [enqueued = self.pullRequests.map { $0.number }] pullRequest in
```

Comment:
imo having such expression in a capture list is kind of confusing. maybe just capture `self.pullRequests` and do `map().contains()`

Reply:
Done in 17b23cc